app/controllers/accounts_controller.rb

```diff
@@ -2,29 +2,18 @@ class AccountsController < ApplicationController
   before_action :authenticate_user!
 
   def new
-    if params[:type].blank? || Account.accountable_types.include?("Account::#{params[:type]}")
-      @account = if params[:type].blank?
-        Account.new
-      else
-        Account.new(accountable_type: "Account::#{params[:type]}")
-      end
-    else
-      head :not_found
-    end
+    @account = build_account
+    head :not_found unless @account
   end
 
-  def show
-  end
+  def show; end
 
   def create
-    @account = Current.family.accounts.build(account_params)
-    @account.accountable = account_params[:accountable_type].constantize.new
+    @account = build_and_associate_account
 
-    if @account.save
-      redirect_to accounts_path, notice: "New account created successfully"
-    else
-      render "new", status: :unprocessable_entity
-    end
+    render "new", status: :unprocessable_entity unless @account.save
```

Comment:
See commend above re modifier clauses.

Reply:
Yes, I responded.

Reply:
The other benefits of the `if...else` style is that it is clearer that the primary function of the action if to save the new account.   By placing the render clause first it reads left to right like the rendering of "new" if the primary function (which it only true in the exceptional case).

Furthermore it is prone to `AbstractController::DoubleRenderError` exceptions (as would be in this particular case if the @account did not save).